registry-authorization-agent/src/main/java/com/hortonworks/registries/schemaregistry/authorization/AuthorizationAgentFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.hortonworks.registries.schemaregistry.authorization;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+public class AuthorizationAgentFactory {
+    public static AuthorizationAgent getAuthorizationAgent(Map<String, Object> props) {
+        if(props == null || props.size() == 0) {
+            return new DummyAuthorizationAgent();
+        }
+
+        String cName = (String) props.get("className");
+        if(cName == null) {
+            return new DummyAuthorizationAgent();
```

Comment:
nit: We can probably combine this "if" with the previous "if" condition as both deal with initialization of DummyAuthorizationAgent with something like this
,,,
   if(props == null || props.size() == 0 || !props.containsKey("className"))
,,,